### ISSUE TITLE
C-CTOR: incorrect usage of term "method"

### ISSUE THREAD

**Issue by jnuss** [rust-lang/api-guidelines#274]
The [C-CTOR ](https://rust-lang.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor)guideline states:

> Constructors are static, inherent methods (C-CCTOR)
> ...
> Constructors are static (no self) inherent methods for the type that they construct...

However, this is a contradiction since a "method" is described in the [reference](https://doc.rust-lang.org/reference/items/associated-items.html#methods) like so:

> "Associated functions whose first parameter is named self are called methods..."

The API guidelines should use consistent terminology as the reference. It is proposed that the incorrect "method" terminology be replaced with the correct "associated function" terminology.

In summary, an idiomatic `new()` constructor is definitely not called a method.

**Comment by jnuss** [comment]
Unfortunately there is another inconsistency in the ecosystem - online docs refer to all associated functions as methods: e.g. https://doc.rust-lang.org/std/boxed/struct.Box.html#method.new

What is considered the correct definition?

**Comment by jnuss** [comment]
The [Rust book](https://doc.rust-lang.org/book/ch05-03-method-syntax.html#associated-functions) appears to agree with the reference:
> All functions defined within an impl block are called associated functions because they’re associated with the type named after the impl. We can define associated functions that don’t have self as their first parameter (and thus are not methods) because they don’t need an instance of the type to work with.

i.e. one more bit of evidence that the API guideline is using incorrect terminology
